typing-react-state/Exercise3/problem/src/App.tsx

```diff
@@ -1,56 +1,56 @@
-// Make changes to the useRefs in order to correct TS errors
-// Take note of the state's structure while adding types.
-// Bonus question - What happens if we use useState
-// instead of useRef?
+/*
 
-import React, { useRef, useState } from "react";
+Exercise:
+
+1) Add types to all of the local state variables.
+2) Bonus: what happens if we replace useRef with useState?
```

Comment:
The changes are great but I really wanted to go over 2 things that's missing here. Firstly, I wanted to avoid using `useState` to get the "Full Name" to show how `useRef` doesn't cause a re-render when `ref` is changed. Right now we use `useState` to create "Full Name" and so I think the 2nd part of the question now has no purpose. 
Secondly, I wanted to show the conditions under which a `ref` is immutable and how we can fix it. This is why I had added 
```
const firstNameRef = useRef<string>(null);
```
This will force the participants to change the `useRef` definition in order to make it mutable. Can we do something about these things?

Reply:
Thanks Avinav, this is helpful feedback. We talked about this during the dry run and I _think_ we wanted to focus on real-world cases where you would need this. This example felt contrived, so I could use some help coming up with a real-world use case to show this.

@CMLabay, I want to make sure this exercise includes what we cover in the training.

Reply:
@avinavdixit25 @CMLabay Could you please take a look at the updated code? I’ve added a `previousFullName` ref that I think covers this feedback.

Reply:
@chasenlehara It looks like this covers it